py/sigver/verify.py

```diff
@@ -1,85 +1,88 @@
     print("          --path=<path>             = Path to verify")
-    print("          --cert=<certname>         = Name of the certificate to be used")
-    print("          --keyvault=<keyvault url> = URL of the keyvault")
+    print("          --cert=<certname>         = (optional) Name of the certificate to be used")
     print("          --sigfile=<filename>      = Signature filename")
     print("")
     sys.exit(0)
 
-
 def main():
-    args = sys.argv[:]
+    """Send REST API request to VerifySignature Azure Function"""
     path = "."
-    key_vault_url = KEY_VAULT_URL
     certificate_name = CERTIFICATE_NAME
     sigfile = "signature.bin"
 
-    args.pop(0)
+    parser = argparse.ArgumentParser(description="Parse arguments")
+    parser.add_argument("--path", default=".",
+                        help="Specify the path. Default is current directory.")
+    parser.add_argument("--cert", default=CERTIFICATE_NAME, help="Specify the certificate name.")
+    parser.add_argument("--sigfile", default="signature.bin", help="Specify the signature file.")
 
-    while args and args[0].startswith("--"):
-        if args[0] == "--help":
-            show_help()
-        if args[0].startswith("--path="):
-            args[0] = args[0].removeprefix("--path=")
-            path = args[0]
-        elif args[0].startswith("--cert="):
-            args[0] = args[0].removeprefix("--cert=")
-            certificate_name = args[0]
-        elif args[0].startswith("--keyvault="):
-            args[0] = args[0].removeprefix("--keyvault=")
-            key_vault_url = args[0]
-        elif args[0].startswith("--sigfile="):
-            args[0] = args[0].removeprefix("--sigfile=")
-            sigfile = args[0]
-        else:
-            print(f"Invalid argument: {args[0]}", file=sys.stderr)
-            sys.exit(1)
+    args = parser.parse_args()
 
-        args.pop(0)
+    path=args.path
+    certificate_name = args.cert
+    sigfile = args.sigfile
 
-    credential = DefaultAzureCredential()
+    digest = base64.b64encode(sha256sum(path, 1024 * 1024, True)).decode('utf-8')
+    if os.path.getsize(sigfile) != 64:
+        LOG.info("Wrong signature size!")
+        return -3
 
-    certificate_client = CertificateClient(
-        vault_url=key_vault_url, credential=credential
-    )
-    key_client = KeyClient(vault_url=key_vault_url, credential=credential)
-
-    certificate = certificate_client.get_certificate(certificate_name)
-
-    print(certificate.name)
+    with open(sigfile, "rb") as file:
+        sig = file.read()
 
-    key = key_client.get_key(certificate_name)
+    signature = base64.b64encode(sig).decode('utf-8')
 
-    crypto_client = CryptographyClient(key, credential)
+    data = {
+        "certificateName": certificate_name, 
+        "Hash": digest,
+        "Signature": signature
+    }
 
-    digest = sha256sum(path, 1024 * 1024, True)
+    headers = {"Content-Type": "application/json"}
+    LOG.info (json.dumps(data))
```

Comment:
Default logging level is WARNING, meaning e.g. LOG.info(...) will never be logged as it now is.
Typically, one would provide an argument (e.g. `-v/--verbose`) to increase the logging verbosity.

Consider adding `--verbose` maybe in a follow-up.